### PR TITLE
Upgrade to Storybook 8

### PR DESCRIPTION
**Pull request by tarjep**
For the most part only followed the `npx storybook@latest upgrade` upgrade guide.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "storybook-theme-switch-addon",
-  "version": "0.0.6",
+  "version": "0.0.7",
```
Comment (krzysztof01-sz) [inline package.json:3]: Increase major version as storybook docs suggests:

https://storybook.js.org/docs/addons/addon-migration-guide#releasing

File: .storybook/main.ts [hunk 2]
```diff
@@ -6,14 +6,12 @@ const config: StorybookConfig = {
...
     "./local-preset.js",
+    "@chromatic-com/storybook",
   ],
   framework: {
     name: "@storybook/react-vite",
     options: {},
   },
-  docs: {
-    autodocs: "tag",
-  },
```
Comment (krzysztof01-sz) [inline .storybook/main.ts:16]: Leave docs object empty

**Comment by krzysztof01-sz** [comment]
Please follow these links and adjust code to these suggestions:

https://storybook.js.org/docs/addons/addon-migration-guide#storybookcomponents-deprecations
https://storybook.js.org/docs/addons/addon-migration-guide#migration-example

**Comment by krzysztof01-sz** [comment]
Fix the issue with `actions.argTypesRegex` value:
https://storybook.js.org/docs/essentials/actions#via-storybooktest-fn-spy-function

**Comment by basher** [comment]
Any progress on this PR?

I use Storybook/HTML v8, with multiple CSS themes, and cannot currently install this addon.

Capture

**Comment by krzysztof01-sz** [comment]
Hi @basher 

I have published a canary version with an upgrade to the newest Storybook version.
Look at this PR: https://github.com/krzysztof01-sz/storybook-theme-switch-addon/pull/2

and test this out:
`npm install storybook-theme-switch-addon@1.0.1--canary.2.19b23d4.0`
`yarn add storybook-theme-switch-addon@1.0.1--canary.2.19b23d4.0`

**Comment by basher** [comment]
Thanks @krzysztof01-sz 

It worked in my local Storybook build! 

There was still a peer dependency issue (I'm running Storybook 8.1.7), so I added `--legacy-peer-deps` flag to resolve it.

My Storybook/HTML now switches between 2 different CSS files.

**Comment by jkazimierczak** [comment]
> There was still a peer dependency issue (I'm running Storybook 8.1.7), so I added --legacy-peer-deps flag to resolve it.

This works, but a sidenote for those who use `pnpm` - it doesn't have an equivalent flag, so you have to set `auto-install-peers` to true:
```
pnpm config set auto-install-peers true --location project
```
This will create `.npmrc` that you should commit, so everyone installing project's dependencies will have the same environment.

**Comment by krzysztof01-sz** [comment]
@jkazimierczak @basher

Thanks for your responses.

#2 have been merged and 1.0.1 version is ready to install. If you will encounter some problems, feel free to open a new issue. You can also bump storybook to the newest version. Upgrade from 8.1.7 to 8.2.5 should not be problematic

**Pull request closed without merging**